elearning-frontend/src/components/admin/CreateWords.js

```diff
@@ -0,0 +1,344 @@
+        })
+    }, [])
+
+    useEffect(() => {
+        if (!props.session) {
+            navigate('/')
+        }
+        else {
+            if(checkAdmin()) {
+                GetCategoryList()
+            }
+            else {
+                navigate('/dashboard')
+            }
+        }
+    }, [props.session])
+
+    // Save word
+    const SaveWord = (e) => {
+        e.preventDefault()
+        saveWord(word, ClearFields)
+    }
+
+    // Clear States
+    const ClearFields = () => {
+        setWord({
+            category_id: '',
+            word: '',
+            choices: {
+                firstChoice: {
+                    choice: '',
+                    is_correct: false
+                },
+                secondChoice: {
+                    choice: '',
+                    is_correct: false
+                },
+                thirdChoice: {
+                    choice: '',
+                    is_correct: false
+                },
+                fourthChoice: {
+                    choice: '',
+                    is_correct: false
+                },
+            },
+            correct_answer: ''
+        })
+    }
+
+    // Set checkbox states
+    const setCorrectAnswer = (index) => {
+        setWord({
+            ...word,
+            choices: {
+                ...word.choices,
+                firstChoice: {
+                    ...word.choices.firstChoice,
+                    is_correct: index === 1 ? true : false
+                },
+                secondChoice: {
+                    ...word.choices.secondChoice,
+                    is_correct: index === 2 ? true : false
+                },
+                thirdChoice: {
+                    ...word.choices.thirdChoice,
+                    is_correct: index === 3 ? true : false
+                },
+                fourthChoice: {
+                    ...word.choices.fourthChoice,
+                    is_correct: index === 4 ? true : false
+                }
+            }
+        })
+    }
+
+    // Set choice field states
+    const setFieldChoiceValue = (index, value) => {
+        setWord({
+            ...word,
+            choices: {
+                ...word.choices,
+                firstChoice: {
+                    ...word.choices.firstChoice,
+                    choice: index === 1 ? value : word.choices.firstChoice.choice
+                },
+                secondChoice: {
+                    ...word.choices.secondChoice,
+                    choice: index === 2 ? value : word.choices.secondChoice.choice
+                },
+                thirdChoice: {
+                    ...word.choices.thirdChoice,
+                    choice: index === 3 ? value : word.choices.thirdChoice.choice
+                },
+                fourthChoice: {
+                    ...word.choices.fourthChoice,
+                    choice: index === 4 ? value : word.choices.fourthChoice.choice
+                }
+            }
+        })
```

Comment:
```suggestion
    const setFieldChoiceValue = (index, value) => {
        setWord({
            ...word,
            choices: {
                word.choices.map(() => {
                  return condition ? correct : incorrect
                })
            }
        })
```

Avoid static way. You can use map.